pkg/controller/controller_instance.go

```diff
@@ -355,27 +354,27 @@ func (c *controller) reconcileServiceInstance(instance *v1alpha1.ServiceInstance
 		return c.pollServiceInstanceInternal(instance)
 	}
 
-	// If there's no async op in progress, determine whether the checksum
-	// has been invalidated by a change to the object. If the instance's
-	// checksum matches the calculated checksum, there is no work to do.
-	// If there's an async op in progress, we need to keep polling, hence
-	// do not bail if checksum hasn't changed.
+	// If there's no async op in progress, determine whether there is a new
+	// generation of the object. If the instance's generation does not match
+	// the reconciled generation, then there is a new generation, indicating
+	// that changes have been made to the instance's spec. If there is an
+	// async op in progress, we need to keep polling, hence do not bail if
+	// there is not a new generation.
 	//
 	// We only do this if the deletion timestamp is nil, because the deletion
 	// timestamp changes the object's state in a way that we must reconcile,
-	// but does not affect the checksum.
+	// but does not affect the generation.
```

Comment:
did you establish whether the generation/userinfo are bumped during a graceful-delete?

Reply:
Generation is not bumped during a delete. PrepareForUpdate is not called.